staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -109,36 +109,9 @@ func newPod() runtime.Object {
 
 func TestCreate(t *testing.T) {
 	ctx, store, etcdClient := testSetup(t)
-
-	key := "/testkey"
-	out := &example.Pod{}
-	obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", SelfLink: "testlink"}}
-
-	// verify that kv pair is empty before set
-	getResp, err := etcdClient.KV.Get(ctx, key)
-	if err != nil {
-		t.Fatalf("etcdClient.KV.Get failed: %v", err)
-	}
-	if len(getResp.Kvs) != 0 {
-		t.Fatalf("expecting empty result on key: %s", key)
-	}
-
-	err = store.Create(ctx, key, obj, out, 0)
-	if err != nil {
-		t.Fatalf("Set failed: %v", err)
-	}
-	// basic tests of the output
-	if obj.ObjectMeta.Name != out.ObjectMeta.Name {
-		t.Errorf("pod name want=%s, get=%s", obj.ObjectMeta.Name, out.ObjectMeta.Name)
-	}
-	if out.ResourceVersion == "" {
-		t.Errorf("output should have non-empty resource version")
-	}
-	if out.SelfLink != "" {
-		t.Errorf("output should have empty selfLink")
-	}
-
-	checkStorageInvariants(ctx, t, etcdClient, store, key)
+	storagetesting.RunTestCreate(ctx, t, store, func(ctx context.Context, t *testing.T, key string) {
+		checkStorageInvariants(ctx, t, etcdClient, store, key)
```

Comment:
nit: I would suggest changing the syntax of checkStorageInvariants to take `Codec` instead of `store` as an argument (so passing store.codec, instead of passing the whole store).

I would make it more clear that the function is actually not messing up anything with the store itself.

WDYT?

Reply:
Seems reasonable, I figured it was also valuable to know that the codec has to be coupled from the store, but it's not meant to be super reusable code anyway.